server/src/test/java/org/opensearch/index/mapper/ObjectMapperTests.java

```diff
@@ -178,6 +178,32 @@ public void testFieldsWithFilledArrayShouldThrowException() throws Exception {
         }
     }
 
+    public void testDottedFields() throws Exception {
+        String mapping = Strings.toString(
+            XContentFactory.jsonBuilder()
+                .startObject()
+                .startObject("properties")
+                .startObject(".")
+                .field("type", "text")
+                .field("dimension", 4)
+                .startObject("method")
+                .field("name", "hnsw")
+                .field("space", "l2")
+                .field("engine", "lucene")
+                .endObject()
+                .endObject()
+                .endObject()
+                .endObject()
+        );
+
+        try {
+            createIndex("test").mapperService().documentMapperParser().parse("tweet", new CompressedXContent(mapping));
+            fail("Expected MapperParsingException");
+        } catch (MapperParsingException e) {
+            assertThat(e.getMessage(), containsString("out of bounds for length"));
```

Comment:
Nitpick - validating against human-readable strings is brittle since these can be easily changed. Please consider something like:

```
assertEquals(e.getCause().getClass(), MapperParsingException.class);
```

Reply:
Sorry, `getCause` isn't applicable here since `ObjectMapper` doesn't wrap the exception. Feel free to ignore this comment.